src/systemdhook.c

```diff
@@ -116,22 +116,26 @@ static int bind_mount(const char *src, const char *dest, int readonly) {
 	return 0;
 }
 
-static int chcon(const char *path, const char *label) {
+static int chperm(const char *path, const char *label, int uid, int gid) {
 	DIR *dir;
 	struct dirent *ent;
 	if ((dir = opendir (path)) != NULL) {
 		/* print all the files and directories within directory */
 		while ((ent = readdir (dir)) != NULL) {
 			_cleanup_free_ char *full_path = NULL;
 			if (asprintf(&full_path, "%s/%s", path, ent->d_name) < 0) {
-				pr_perror("Failed to create path for chcon");
+				pr_perror("Failed to create path for chperm");
 				closedir(dir);
 				return -1;
 
 			}
 			if (setfilecon (full_path, label) < 0) {
 				pr_perror("Failed to set context %s on %s", label, path);
 			}
+			/* Change uid and gid to something the container can handle */
+			if (chown(full_path, uid, gid) == -1) {
+				pr_perror("Failed to chown %d,%d to full_path owner: %s, errno: [%d]", uid, gid, full_path, errno);
+			}
```

Comment:
systemd-named-path is set to something like: /var/lib/docker/808080.808080/devicemapper/mnt/e18049df157e4fad7f124aad7d3ab4f20d4e272329128891f50190870e3dc568/rootfs//sys/fs/cgroup/systemd/system.slice/docker-a6b4e4a567112b0649dcd3d9b8cf087b9ab7e708c1f9281eaaf3004affd36101.scope

Seen the second line from journalctl below.

May 26 10:29:09 ip-172-18-11-55.ec2.internal oci-systemd-hook[2476]: systemdhook <error>: Failed to set context  on /var/lib/docker/808080.808080/devicemapper/mnt/e18049df157e4fad7f124aad7d3ab4f20d4e272329128891f50190870e3dc568/rootfs//sys/fs/cgroup/systemd/system.slice/docker-a6b4e4a567112b0649dcd3d9b8cf087b9ab7e708c1f9281eaaf3004affd36101.scope: Read-only file system
May 26 10:29:09 ip-172-18-11-55.ec2.internal oci-systemd-hook[2476]: systemdhook <error>: Failed to chown 808080,808080 to full_path owner: /var/lib/docker/808080.808080/devicemapper/mnt/e18049df157e4fad7f124aad7d3ab4f20d4e272329128891f50190870e3dc568/rootfs//sys/fs/cgroup/systemd/system.slice/docker-a6b4e4a567112b0649dcd3d9b8cf087b9ab7e708c1f9281eaaf3004affd36101.scope/.., errno: [30]: Read-only file system